src/server/kernel/wlayersurface.h

```diff
@@ -0,0 +1,113 @@
+// Copyright (C) 2023 rewine <luhongxu@deepin.org>.
+// SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+
+#pragma once
+
+#include <WSurface>
+#include <wtoplevelsurface.h>
+
+struct wlr_layer_surface_v1;
+
+QW_BEGIN_NAMESPACE
+class QWSurface;
+class QWLayerSurfaceV1;
+QW_END_NAMESPACE
+
+WAYLIB_SERVER_BEGIN_NAMESPACE
+
+class WSeat;
+class WLayerSurfacePrivate;
+class WAYLIB_SERVER_EXPORT WLayerSurface : public WToplevelSurface, public WObject
+{
+    Q_OBJECT
+    W_DECLARE_PRIVATE(WLayerSurface)
+    //Q_PROPERTY(WLayerSurface* parentXdgSurface READ parentXdgSurface NOTIFY parentXdgSurfaceChanged FINAL)
+    Q_PROPERTY(WSurface* surface READ surface NOTIFY surfaceChanged)
+    Q_PROPERTY(LayerType layer READ layer NOTIFY layerChanged)
+    Q_PROPERTY(uint32_t ancher READ ancher NOTIFY ancherChanged)
+    Q_PROPERTY(int32_t exclusiveZone READ exclusiveZone NOTIFY keyboardInteractivityChanged)
+    Q_PROPERTY(int32_t leftMargin READ leftMargin NOTIFY leftMarginChanged)
+    Q_PROPERTY(int32_t rightMargin READ rightMargin NOTIFY rightMarginChanged)
+    Q_PROPERTY(int32_t topMargin READ topMargin NOTIFY topMarginChanged)
+    Q_PROPERTY(int32_t bottomMargin READ bottomMargin NOTIFY bottomMarginChanged)
+    Q_PROPERTY(KeyboardInteractivity keyboardInteractivity READ keyboardInteractivity NOTIFY keyboardInteractivityChanged)
+
+    QML_NAMED_ELEMENT(WaylandLayerSurface)
+    QML_UNCREATABLE("Only create in C++")
+
+public:
+    explicit WLayerSurface(QW_NAMESPACE::QWLayerSurfaceV1 *handle, QObject *parent = nullptr);
+    ~WLayerSurface();
+
+    enum LayerType {
+        LayerBackground = 0,
+        LayerBottom = 1,
+        LayerTop = 2,
+        LayerOverlay = 3,
+    };
+    Q_ENUM(LayerType)
+
+    enum AnchorType {
+        AnchorNone = 0,
+        AnchorTop = 1,
+        AnchorBottom = 2,
+        AnchorLeft = 4,
+        AnchorRight = 8,
+    };
+    Q_ENUM(AnchorType)
+
+    enum KeyboardInteractivity {
+        KeyboardInteractivityNone = 0,
+        KeyboardInteractivityExclusive = 1,
+        KeyboardInteractivityOnDemand = 2,
+    };
+    Q_ENUM(KeyboardInteractivity)
+
+
+    bool isPopup() const;
+    bool doesNotAcceptFocus() const override;
+    bool isActivated() const override;
+    WSurface *surface() const override;
+    QW_NAMESPACE::QWLayerSurfaceV1 *handle() const;
+    wlr_layer_surface_v1 *nativeHandle() const;
+    QW_NAMESPACE::QWSurface *inputTargetAt(QPointF &localPos) const;
+
+    static WLayerSurface *fromHandle(QW_NAMESPACE::QWLayerSurfaceV1 *handle);
+    static WLayerSurface *fromSurface(WSurface *surface);
+
+    QRect getContentGeometry() const;
+    int keyboardFocusPriority() const override;
```

Comment:
enum?

Reply:
`keyboardInteractivity` is enum, but `keyboardFocusPriority` should just a int